baseplate/diagnostics/metrics.py

```diff
@@ -48,6 +48,10 @@ def on_child_span_created(self, span):  # pragma: nocover
         observer = MetricsSpanObserver(self.batch, "clients." + span.name)
         span.register(observer)
 
+    def on_local_span_created(self, context, span, component_name):  # pragma: nocover
+        observer = MetricsSpanObserver(self.batch, component_name + "." + span.name)
```

Comment:
Given the issue with the sqlalchemy guy, maybe we can just rely on checking types of spans for generating the namespace? then we probably don't need a separate observer method for just local spans which frees up the class hierarchy a bit.